internal/bson_value_util.go

```diff
@@ -0,0 +1,74 @@
+package internal
+
+import (
+	"fmt"
+	"reflect"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/bsoncodec"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+type ErrNilValue struct{}
```

Comment:
AFAIK there is no way to move the transformValue function without also moving the underlying errors and relying on the "mongo.replaceErrors" function to assign mongo errors to the underlying internal errors.